pkg/ruler/ruler.go

```diff
@@ -458,6 +458,7 @@ func (r *Ruler) run(ctx context.Context) error {
 	}
 }
 
+// It's not safe to call this function concurrently.
```

Comment:
nit: Can you expand on this comment a bit? Maybe something like "we expect this is only called from the service `run` method"